app/controllers/events_controller.rb

```diff
@@ -7,11 +7,20 @@ class EventsController < ApplicationController
 
   # GET /events
   def index
-    @events = Event.sorted_by_start_date(!can?(:view_unpublished, Event))
+    @events = Event.sorted_by_start_date(false)
+    if !can? :view_hidden, Event
+      @events = @events.select{|event| event.hidden == false}
+    end
+    if !can? :view_unpublished, Event
```

Comment:
> This is unrelated to the PR, right? I assume this was a bug?
the unpuplished events were filtered before by the sorted_by_start_date function (so this is already tested in spec/features/event_spec.rb:28). 
Because i wanted to distinguish between hidden and unpuplished events, I test the rigths to view them seperated. I know at the moment, this distinction makes no difference, but i don't think it is a good idea to summerize it to one ability.